### PR TITLE
GH-7: Add macOS, Windows CI

### LINKED ISSUE #7: Add test CI: macOS
This part: https://github.com/apache/arrow/blob/a6fe59592c643d591ee719041dd2b5d8b9ada7d6/.github/workflows/java.yml#L113-L142

### LINKED ISSUE #8: Add test CI: Windows
This part: https://github.com/apache/arrow/blob/a6fe59592c643d591ee719041dd2b5d8b9ada7d6/.github/workflows/java.yml#L144-L173

### PR DESCRIPTION
**Pull request by lidavidm**
Fixes #7.
Fixes #8.

### CODE REVIEW

**Comment by lidavidm** [comment]
Hmm, macOS `realpath` is not as capable

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -81,12 +81,13 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        - arch: AMD64
-          jdk: 11
-          macos: 13
-        - arch: AArch64
-          jdk: 11
-          macos: latest
+        include:
```
Comment (raulcd) [inline .github/workflows/test.yml:84]: oops, I missed that :)

Comment (lidavidm) [reply .github/workflows/test.yml:84]: Updated, tests all pass

**Review by kou — APPROVED** [review]
+1

**Pull request merged**